packages/common-lib/src/services/commentRegistryService.js

```diff
@@ -0,0 +1,88 @@
+import { get, post, update as coreUpdate } from './RestClient'
+import mapInterfaceData from './mapInterfaceData'
+import manifest from '../manifest.json'
+
+const interfaceData = {
+  id: 'commentId',
+  contextId: 'contextId',
+  context: 'context',
+  userId: 'userId',
+  comment: 'comment',
+  parentId: 'parentId',
+  status: 'status',
+  privacy: 'privacy'
+}
+
+let only = Object.keys(interfaceData)
+
+export const getAll = async ({ limit, ...params } = {}, header = {}) => {
+  let headers = {
+    ...header,
+    headers: {
+      ...header.header,
+      Authorization: 'Bearer ' + localStorage.getItem('token')
+    }
+  }
+  const result = await post(
+    manifest.api_url + '/comment/search',
+    { filters: params, limit: limit },
+    {
+      headers: headers?.headers ? headers?.headers : {}
+    }
+  )
+  if (result.data.data) {
+    return result.data.data.map((e) => mapInterfaceData(e, interfaceData))
+  } else {
+    return []
+  }
+}
+
+export const create = async (data, headers = {}) => {
+  let newInterfaceData = interfaceData
+  let header = {
+    ...headers,
+    Authorization: 'Bearer ' + localStorage.getItem('token')
+  }
+  newInterfaceData = {
+    ...interfaceData,
+    removeParameter: headers?.removeParameter ? headers?.removeParameter : [],
+    onlyParameter: headers?.onlyParameter ? headers?.onlyParameter : only
+  }
+  let newData = mapInterfaceData(data, newInterfaceData, true)
+  const result = await post(manifest.api_url + '/comment', newData, {
+    headers: header
+  })
+  if (result.data) {
+    let { Comment } = result.data?.data?.result
```

Comment:
result.data?.data?.result looks strange

Reply:
This is response coming from back-end

Reply:
what is response for this API call? can we fix the backend response?

Reply:
Response body

{
  "statusCode": 200,
  "message": "Ok.",
  "data": [
    {
      "commentId": "faad8d77-1644-42c8-a62d-5616df82b80d",
      "contextId": "eef7ee68-29f1-4936-9d00-6b457c94f364",
      "context": "Worksheet",
      "userId": "1cd90456-e095-4f03-88e7-4edda137ca4b",
      "comment": "hello",
      "privacy": "",
      "parentId": "",
      "status": "",
      "createdAt": "2022-06-28T06:41:25.189Z",
      "updatedAt": "2022-06-28T06:41:25.189Z",
      "createdBy": "anonymous",
      "updatedBy": "anonymous"
    },
    {
    ...

Reply:
const { osid } = await commentRegistryService.create(newData);